lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -1078,6 +1092,56 @@ static void print(OpAsmPrinter &printer, rtl::StructInjectOp op) {
   printer << " : " << op.input().getType();
 }
 
+//===----------------------------------------------------------------------===//
+// UnionCreateOp
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseUnionCreateOp(OpAsmParser &parser,
+                                      OperationState &result) {
+  UnionType declType;
+  StringAttr field;
+  OpAsmParser::OperandType input;
+  llvm::SMLoc fieldLoc = parser.getCurrentLocation();
+
+  if (parser.parseAttribute(field, "field", result.attributes) ||
+      parser.parseComma() || parser.parseOperand(input) ||
+      parser.parseOptionalAttrDict(result.attributes) ||
+      parser.parseColonType(declType))
+    return failure();
+
+  Type inputType = declType.getFieldType(field.getValue());
+  if (!inputType) {
+    parser.emitError(fieldLoc, "cannot find union field '")
+        << field.getValue() << "'.";
```

Comment:
nit: please remove the period from the end of the diagnostic.  Compiler errors don't have them by tradition.